Hold-Krykke/Security_and_FullstackJavascript_Exam
# Initial Backend Setup

_Typescript, Express, NodeJS, GraphQL, [Apollo](https://www.apollographql.com/), [Google-Auth](https://www.npmjs.com/package/google-auth-library)_ 

**How to use for reviewing**
1. Check out branch 
2. `npm install`
3. `npm run start:dev`
4. Open a browser and go to: [localhost:3000/graphql](http://localhost:3000/graphql)
5. Make a query: `{helloWorld}`
6. Report back if it works or not. 

**Discussion:**
- Need help on the tsconfig - Not sure how it should be configured
   - Maybe we should comment in the sourcemap for debugging purposes? 
   - Notice strict is true. Should it be? 

**How it looks on my machine:**
![image](https://user-images.githubusercontent.com/11192971/81845827-69509600-9551-11ea-8c61-108135e2912b.png)

**[Links](https://www.apollographql.com/)**
[Apollo Server Express npmjs](https://www.npmjs.com/package/apollo-server-express)
[Getting Started Apollo Server](https://www.apollographql.com/docs/apollo-server/ )
[Schema](https://www.apollographql.com/docs/apollo-server/schema/schema/)
[Resolvers](https://www.apollographql.com/docs/apollo-server/data/resolvers/)
[No Looping in Queries](https://www.npmjs.com/package/graphql-depth-limit)
[Apollo Client](https://www.apollographql.com/docs/react/)
[Apollo Server](https://www.apollographql.com/docs/apollo-server/)

**Setup Overview**
![image](https://user-images.githubusercontent.com/11192971/81853099-4d9ebd00-955c-11ea-87b2-3168ac67f824.png)
